torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1344,6 +1344,44 @@ def cast_symbool_to_symint_guardless(symbool: torch.SymBool) -> torch.SymInt:
     int_sym = sympy.Piecewise((1, symbool.node.expr), (0, True))
     return symbool.node.shape_env.create_symintnode(int_sym, hint=int(symbool.node.require_hint()))
 
+def sym_ite(pred, then_val, else_val):
+    if isinstance(pred, SymBool):
+        if pred.node.is_constant():
+            constant_pred = guard_bool(pred)
+            return then_val if constant_pred else else_val
+        return _sym_ite(pred, then_val, else_val)
+    elif isinstance(pred, bool):
+        return then_val if pred else else_val
+    else:
+        raise NotImplementedError(f"sym_ite only supports bool or SymBool pred but got {type(pred)}.")
+
+def _sym_ite(pred, then_val, else_val):
+    pred_node = pred.node
```

Comment:
Once you go through C++, it will no longer be necessary to convert these to nodes

Reply:
Why is that? Do the python implementation of sym ops call the c++ implementation? Can I have a code pointer?